cmd/delete.go

```diff
@@ -117,19 +120,22 @@ func runDeleteBefore(cmd *cobra.Command, args []string) {
 		tracelog.ErrorLogger.FatalError(err)
 	}
 	before, err := time.Parse(time.RFC3339, beforeStr)
-	if err != nil {
-		target, err := internal.FindTargetBeforeName(folder, beforeStr, modifier)
-		// after updating delete before time it will be made common
+	if err == nil {
+		storage.SetLessFunction(earlierCreated) // it needs for sort []storage.Object
+		potentialTarget, err := internal.FindFirstLaterOrEqualTime(folder, before)
 		if err != nil {
 			tracelog.ErrorLogger.FatalError(err)
 		}
-		if confirmed {
-			err = storage.DeleteObjectsWhere(folder, func(object storage.Object) bool {
-				return earlierCreated(object, target)
-			})
-		}
-	} else {
-		internal.HandleDeleteBeforeTime(folder, before, modifier, !confirmed)
+		beforeStr = fetchBackupName(potentialTarget)
+	}
+	target, err := internal.FindTargetBeforeName(folder, beforeStr, modifier)
```

Comment:
What about replacing this function with a function FindTargetLSNBeforeName? It seems to me that such a substitution will simplify the code.

Reply:
I thought about it, but it is postgres specified logic.